kernel/src/hil/sensors.rs

```diff
@@ -31,6 +31,47 @@ pub trait HumidityClient {
     fn callback(&self, value: usize);
 }
 
+/// A basic interface for a Air Quality sensor
+pub trait AirQualityDriver<'a> {
+    /// Set the client to be notified when the capsule has data ready.
+    fn set_client(&self, client: &'a dyn AirQualityClient);
+
+    /// Read the CO2 or equivalent CO2 (eCO2) from the sensor.
+    /// This will trigger the `AirQualityClient` `co2_data_available()`
+    /// callback when the data is ready.
+    ///
+    /// This function might return the following errors:
+    /// - `BUSY`: Indicates that the hardware is busy with an existing
+    ///           operation or initilisation/calibration.
+    /// - `NOSUPPORT`: Indicates that this data type isn't supported.
+    fn read_co2(&self) -> Result<(), ErrorCode>;
+
+    /// Read the Total Organic Compound (TVOC) from the sensor.
+    /// This will trigger the `AirQualityClient` `tvoc_data_available()`
+    /// callback when the data is ready.
+    ///
+    /// This function might return the following errors:
+    /// - `BUSY`: Indicates that the hardware is busy with an existing
+    ///           operation or initilisation/calibration.
+    /// - `NOSUPPORT`: Indicates that this data type isn't supported.
+    fn read_tvoc(&self) -> Result<(), ErrorCode>;
+}
+
+/// Client for receiving Air Quality readings
+pub trait AirQualityClient {
+    /// Called when a CO2 or equivalent CO2 (eCO2) reading has completed.
+    ///
+    /// - `value`: will contain the latest CO2 reading in ppm. An example value
+    ///            might be `400`.
+    fn co2_data_available(&self, value: u32);
```

Comment:
Why is CO2 in ppm but VOC in ppb? Should they both be ppb (given a 32-bit value)?

Reply:
I think it makes sense to use these units, since normal outdoor eCO2 ranges are <~1000ppm. But the TVOC typically can be in the range of ~200-5500ppb. So for instance, 500ppb = 0.5ppm. Avoids having to deal with floats, if we use ppb.

Reply:
I think given the ranges in which these units are measures it makes sense?
eCO2 usually <~1000ppm and tVOC is typically like 200-5500ppb.

Reply:
What @twilfredo said. It's to do with the usual ranges. It matches what I see others using and what the datasheets supply formulas to calculate